File: demo_search.py
import os

from rjh_agent import AgentConfig, HitlConfig, SandboxConfig, create_rjh_agent
from rjh_agent.env import load_dotenv
from rjh_agent.tools import TavilySearch


def main() -> None:
    load_dotenv()

    tavily_key = os.getenv("TAVILY_API_KEY")
    if not tavily_key:
        raise RuntimeError("缺少 TAVILY_API_KEY，请在 .env 中配置")

    search = TavilySearch(tavily_key)
    agent = create_rjh_agent(
        AgentConfig(
            name="RJH OpenCodex (DeepSeek) - 搜索版",
            skills_dir=".rjh/skills",
            sandbox=SandboxConfig(workspace_path=os.getcwd(), output_dir="output", verbose=True),
            hitl=HitlConfig(enabled=False),
            system_prompt="""
你是一个具备网络搜索能力的 AI 助手，由 DeepSeek 驱动。
需要写文件时请用：
```filename:文件名.md
内容
```
""",
        )
    )

    query = "LangChain Deep Agent 快速入门教程 2026"
    print(f"\n搜索关键词：{query}")

    search_results = search.search(query)
    print(f"[Search] 获取到 {len(search_results)} 条结果")

    joined_results = "\n---\n".join(
        f"[{index}] {item.title}\n来源：{item.url}\n摘要：{item.content}"
        for index, item in enumerate(search_results, start=1)
    )

    user_message = f"""
请根据以下搜索结果，总结 LangChain Deep Agent 的核心概念和快速入门步骤，
输出一份清晰的中文学习笔记，并保存到文件 deep-agent-notes.md。

搜索结果：
{joined_results}

要求：
1. 用简洁的中文总结
2. 包含核心概念解释
3. 包含快速入门步骤
4. 写入文件 deep-agent-notes.md
"""

    result = agent.invoke_stream(user_message)

    if not result.files_written:
        sandbox = agent.get_sandbox()
        if sandbox:
            sandbox.write_file("deep-agent-notes.md", result.content)
            print("\n已手动保存结果到 output/deep-agent-notes.md")


if __name__ == "__main__":
    main()

File: src/demo-search.ts
/**
 * Demo 2：搜索 + 写文件完整链路
 * 需要：TAVILY_API_KEY 环境变量
 * 运行命令：npm run demo:search
 */

import 'dotenv/config'
import { createRJHAgent } from './agent.js'
import { TavilySearch } from './tools/tavily-search.js'

async function main() {
  if (!process.env.TAVILY_API_KEY) {
    console.error('❌ 缺少 TAVILY_API_KEY，请在 .env 中配置')
    console.log('   获取地址：https://app.tavily.com/ (免费额度每月 1000 次)')
    process.exit(1)
  }

  const search = new TavilySearch(process.env.TAVILY_API_KEY)

  const agent = await createRJHAgent({
    name: '智能体 OpenCodex (DeepSeek) - 搜索版',
    skillsDir: '.rjh/skills',
    sandbox: { workspacePath: process.cwd(), outputDir: 'output', verbose: true },
    hitl: { enabled: false },
    systemPrompt: `
你是一个具备网络搜索能力的 AI 助手，由 DeepSeek 驱动。
需要写文件时请用：
\`\`\`filename:文件名.md
内容
\`\`\`
`,
  })

  const query = 'LangChain Deep Agent 快速入门教程 2026'
  console.log(`\n 搜索关键词：${query}`)

  const searchResults = await search.search(query)
  console.log(`[Search] 获取到 ${searchResults.length} 条结果`)

  const userMessage = `
请根据以下搜索结果，总结 LangChain Deep Agent 的核心概念和快速入门步骤，
输出一份清晰的中文学习笔记，并保存到文件 deep-agent-notes.md。

搜索结果：
${searchResults.map((r, i) => `[${i + 1}] ${r.title}\n来源：${r.url}\n摘要：${r.content}`).join('\n---\n')}

要求：
1. 用简洁的中文总结
2. 包含核心概念解释
3. 包含快速入门步骤
4. 写入文件 deep-agent-notes.md
`

  const result = await agent.invokeStream(userMessage)

  if (result.filesWritten.length === 0) {
    const sandbox = agent.getSandbox()
    if (sandbox) {
      sandbox.writeFile('deep-agent-notes.md', result.content)
      console.log('\n✅ 已手动保存结果到 output/deep-agent-notes.md')
    }
  }
}

main().catch(console.error)
